tests/kernel/wave/attention/alibi_attention_test.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2025 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import pytest
+import torch
+import math
+import iree.turbine.kernel as tk
+from iree.turbine.kernel.lang.global_symbols import *
+from iree.turbine.kernel.wave.utils import (
+    get_default_run_config,
+    get_default_scheduling_params,
+    device_arange,
+    device_full,
+    device_randn,
+    device_zeros,
+    to_default_device,
+)
+from iree.turbine.kernel.wave.constraints import MMAType
+from iree.turbine.kernel.wave.templates.alibi_attention import (
+    get_alibi_attention_kernel,
+)
+from iree.turbine.kernel.wave.templates.attention_common import (
+    AttentionShape,
+)
+import os
+from torch.testing import assert_close
+from ..common.utils import (
+    require_e2e,
+    require_cdna3,
+    enable_scheduling_barriers,
+    dump_generated_mlir,
+    get_default_arch,
+)
+from ..common.shapes import get_test_shapes
+from typing import List, Optional, Tuple
+
+shapes = [(128, 128, 128, 128, 128, 128)]
+
+
+def get_relative_positions(
+    seq_len: int, kv_seq_len: Optional[int] = None
+) -> torch.Tensor:
+    """Returns a lower-trinagular tensor with distance between rows and columns.
+
+    The tensor resembles the following:
+
+        [ 0  0  0  0  0]
+        [-1  0  0  0  0]
+        [-2 -1  0  0  0]
+        [-3 -2 -1  0  0]
+    """
+    if not kv_seq_len:
+        kv_seq_len = seq_len
+    x = torch.arange(kv_seq_len)[None, :]
+    y = torch.arange(seq_len)[:, None]
+    return to_default_device(torch.minimum(x - y, torch.zeros(seq_len, kv_seq_len)))
+
+
+def precompute_alibi_slopes(n_heads: int) -> torch.Tensor:
+    """Computes the constant slopes of linear biases to be added to the attention scores."""
+    n = 2 ** math.floor(math.log2(n_heads))
+    m_0 = 2.0 ** (-8.0 / n)
+    m = torch.pow(m_0, torch.arange(1, 1 + n))
+    if n < n_heads:
+        m_hat_0 = 2.0 ** (-4.0 / n)
+        m_hat = torch.pow(m_hat_0, torch.arange(1, 1 + 2 * (n_heads - n), 2))
+        m = torch.cat([m, m_hat])
+    return to_default_device(m)
+
+
+def validate_accuracy(
+    query: torch.Tensor, key: torch.Tensor, value: torch.Tensor, output: torch.Tensor
+) -> torch.Tensor:
+    # Precompute values.
+    dk_sqrt = math.sqrt(1.0 / query.shape[-1])
+    alibi_slopes = precompute_alibi_slopes(query.shape[0])
+
+    # Straightforward implementation of attention with bias.
+    scores = torch.matmul(query, key.transpose(-1, -2)) * dk_sqrt
+    bias = alibi_slopes.unsqueeze(-1).unsqueeze(-1) * get_relative_positions(
```

Comment:
We don't need to change for this PR, but for future PRs on `flexattention` related stuff, we can also just import flex_attention and use it directly, as seen also in https://github.com/iree-org/iree-turbine/commit/176d443b0256eac484606bf7fc8cebbea29e2e0b#diff-1f375ba26b8000b24b7d3a662f186833ab26f2a9782dbe099b931fc3263445aeR83